src/zenml/client.py

```diff
@@ -2236,6 +2249,69 @@ def get_run_step(self, step_run_id: UUID) -> StepRunResponseModel:
         """
         return self.zen_store.get_run_step(step_run_id)
 
+    # -------------
+    # - Artifacts -
+    # -------------
+
+    def list_artifacts(
+        self,
+        artifact_uri: Optional[str] = None,
+        artifact_store_id: Optional[UUID] = None,
+    ) -> List[ArtifactResponseModel]:
+        """Get all artifacts.
+
+        Args:
+            artifact_uri: If provided, only return artifacts with this URI.
+            artifact_store_id: If provided, only return artifacts from this
+                artifact store.
+
+        Returns:
+            A list of artifacts.
+        """
+        return self.zen_store.list_artifacts(
+            artifact_uri=artifact_uri,
+            artifact_store_id=artifact_store_id,
```

Comment:
In order to be able to use this type of filtering, users would need to know the exact URI or the exact id of the artifact store.  Do you think that it would help to include any type of prefix matching here (either with the URI or the artifact store)? 

This might even enable the CLI as follows:

Instead of:
``` 
zenml artifact list --artifact_uri=/the/exact/uri.csv --artifact_store_id=ThE-3xaC4-UR1
```

this:

```
zenml artifact list --artifact_uri=/the/root/of/all/trainer/step/artifact/ouptuts --artifact_store=NAME
```

Reply:
We could do it but IMO it's not worth the effort, since I imagine once you have a couple of runs, browing artifacts in the CLI will not be the best experience anyways. Perhaps @AlexejPenner 's filtering PR will make this easier?

Reply:
yeah I think this will be already dooable on the filtering branch

Reply:
Let's revisit this after the filtering branch then